### PR TITLE
Create support-cycle.md

### PR DESCRIPTION
**Pull request by keithamus**
This document codifies some official support guidelines around our release cycle, and platform support - for all chai projects. @chaijs/core please take the time to go through this and let me know if any changes are needed. Happy to discuss any detail of this - this is not a dictatorial document, merely just a starting point for what I consider sensible.

### CODE REVIEW

File: support-cycle.md
Comment (vieiralucas) [inline support-cycle.md]: I think that this have a typo
Shouldn't it be `the latest version *of* Chrome` instead of `the latest version *so* Chrome`?

File: support-cycle.md
Comment (vieiralucas) [inline support-cycle.md]: I think that this fails to say which versions will be supported by chai
I can infer that it is the last 3 releases, but it is not explicitly said as in Firefox for example:
`For us this means we essentially support a years worth of Firefox releases.`

File: support-cycle.md
Comment (vieiralucas) [inline support-cycle.md]: Maybe this is just me (and I'm not a native english speaker), but when I read `October 25` I think of the 25th day of October. Maybe this could be reword to `October of 2025`?

**Review by vieiralucas — CHANGES_REQUESTED** [review]
This is looking awesome to me ❤️

**Comment by keithamus** [comment]
@vieiralucas corrected those errors and further clarified our support cycle with more concrete estimations of versions we should support - up until 2020. Hopefully this gives us very clear guidelines of what our support should look like now.

**Review by vieiralucas — APPROVED** [review]

**Comment by keithamus** [comment]
Can we get one or two more reviews from @chaijs/core?

File: support-cycle.md
Comment (meeber) [inline support-cycle.md]: On line 29, is the date supposed to be `2019-04-01`?

File: support-cycle.md
Comment (meeber) [inline support-cycle.md]: Nits:

- I think "organizations" should be singular here
- "documented below"

File: support-cycle.md
Comment (meeber) [inline support-cycle.md]: "Edge release cycle" instead of Safari

File: support-cycle.md
Comment (meeber) [inline support-cycle.md:137]: So is our official stance to no longer support IE 10? Mostly thinking about this: https://github.com/chaijs/chai/issues/860

File: support-cycle.md
Comment (meeber) [inline support-cycle.md:146]: Ouch

File: support-cycle.md
Comment (meeber) [inline support-cycle.md]: Just noting that personally I probably won't be able to do something like this, even if money is on the table.

File: support-cycle.md
Comment (vieiralucas) [reply support-cycle.md:146]: 🐼 🔫

File: support-cycle.md
Comment (keithamus) [reply support-cycle.md:137]: I don't have hugely strong opinions on whether we keep supporting IE10. I'm happy to drop it, I'm happy to keep supporting it. Mostly, decisions in this initial draft were guided by vendor support and not set in stone; the intent was to document produce _some decision_ which can trigger a conversation and we can settle on something else should we need to (to which I guess this has served it's purpose). 

To try and input into a discussion of supporting IE10, let's state some facts:

 - This document encodes that Chai 4 will continue to support IE 10
 - This document encodes that all Chai projects will support IE 11 until 2025, its EOL date.

Now some questions: Should we just not change this section, and have it as an implicit guide that we drop IE10 for Chai 5? Conversely, would it be feasible to _support_ IE10 _and_ IE11 to 2025? These I think are the two extremes we have to chose from, is there a sensible non-arbitrary choice in between those?

File: support-cycle.md
Comment (keithamus) [reply support-cycle.md]: Again, this was just here to trigger the discussion. I felt the need to document this as a way to rebuff support requests that are outside of our remit. 

Is this something we feel comfortable drawing a line in the sand about? I understand our various _real_ jobs might prevent some of us from committing to doing external paid work, or even just having the flexibility for time off. I certainly don't want to promote a position for those who have the privilege of flexibility/agency to benefit over those that don't, but I am also very much keen on seeing the maintainership of Chai being rewarded more concretely (fiscally) for your hard work. 

I would definitely like to hear more people's thoughts on all of this. Perhaps this section shouldn't be included in this document? Perhaps we should take a harder line and simple document that none of this work should be undertaken?

**Comment by vieiralucas** [comment]
Hey @keithamus is this ready?
I really want to get rid of IE 10 😄

**Comment by keithamus** [comment]
I've updated this doc to reflect today's platform versions, and also remove the guidelines around paid support. Would be happy if folks could read through and ✅ or ❌ accordingly 😄

File: support-cycle.md
Comment (lucasfcosta) [reply support-cycle.md:137]: I think we can simply drop IE 10, can't we?

I remember we have had some discussion around this in this issue (https://github.com/chaijs/chai/issues/860) and considering what has been said there it would be better to drop it.

**Review by lucasfcosta — APPROVED** [review]
LGTM. I only had a comment on IE10 support. I'd rather drop it considering the mentioned issue, but I won't block merging if people think otherwise.

**Comment by keithamus** [comment]
> I'd rather drop [IE10]

Perhaps I can word it better, but the intention of this document _is_ to drop IE10 (but still support 11):

> [...] expect to support Internet Explorer 11 until then. We do not support older versions of Internet Explorer.

**Review by vieiralucas — APPROVED** [review]
LGTM 👏

**Pull request merged**
